x/mint/keeper/keeper.go

```diff
@@ -67,21 +70,20 @@ func NewKeeper(
 	return k
 }
 
-// GetAuthority returns the x/mint module's authority.
-func (k Keeper) GetAuthority() string {
-	return k.authority
-}
+// SetMintFn sets the mint function.
```

Comment:
That's meh, why not add this in the constructor?

Reply:
then the mint module would take the staking keeper in the constructor, i wanted to avoid the staking keeper being required in the constructor to avoid optional parameters in the constructor. Also we dont know if the mintFn will require the mint module before its past to the constructor. It would be a weird look.

Reply:
Still feels like a code smell. `types.MintFn` doesn't look like it requires the staking module.
That function only has access to state via the environment, how would you use staking or mint in there? Via the router? Then you already don't need the keepers.

Reply:
the default function requires staking and mint keeper. that is the caveat

Reply:
I see, I think we can still make it okay with depinject. Posted a comment below

Reply:
made it optional for depinject